src/appkit.cs

```diff
@@ -27427,4 +27579,279 @@ interface NSStepperTouchBarItem
+
+		[Export ("textCompletionType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType TextCompletionType { get; set; }
+	}
+
+	interface INSTextCheckingClient { }
+
+	[Protocol]
+	[Mac (10,15)]
+	interface NSTextCheckingClient : NSTextInputTraits, NSTextInputClient
+	{
+		[Abstract]
+		[Export ("annotatedSubstringForProposedRange:actualRange:")]
+		[return: NullAllowed]
+		NSAttributedString GetAnnotatedSubstring (NSRange range, [NullAllowed] ref NSRange actualRange);
+
+		[Abstract]
+		[Export ("setAnnotations:range:")]
+		void SetAnnotations (NSDictionary<NSString, NSString> annotations, NSRange range);
+
+		[Abstract]
+		[Export ("addAnnotations:range:")]
+		void AddAnnotations (NSDictionary<NSString, NSString> annotations, NSRange range);
+
+		[Abstract]
+		[Export ("removeAnnotation:range:")]
+		void RemoveAnnotation (string annotationName, NSRange range);
+
+		[Abstract]
+		[Export ("replaceCharactersInRange:withAnnotatedString:")]
+		void ReplaceCharacters (NSRange range, NSAttributedString annotatedString);
+
+		[Abstract]
+		[Export ("selectAndShowRange:")]
+		void SelectAndShow (NSRange range);
+
+		[Abstract]
+		[Export ("viewForRange:firstRect:actualRange:")]
+		[return: NullAllowed]
+		NSView GetView (NSRange range, [NullAllowed] ref CGRect firstRect, [NullAllowed] ref NSRange actualRange);
+
+		[Abstract]
+		[NullAllowed, Export ("candidateListTouchBarItem")]
+		NSCandidateListTouchBarItem CandidateListTouchBarItem { get; }
+	}
+
+	[Mac (10,15)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface NSWorkspaceOpenConfiguration : NSCopying
+	{
+		[Static]
+		[Export ("configuration")]
+		NSWorkspaceOpenConfiguration Create ();
+
+		[Export ("promptsUserIfNeeded")]
+		bool PromptsUserIfNeeded { get; set; }
+
+		[Export ("addsToRecentItems")]
+		bool AddsToRecentItems { get; set; }
+
+		[Export ("activates")]
+		bool Activates { get; set; }
+
+		[Export ("hides")]
+		bool Hides { get; set; }
+
+		[Export ("hidesOthers")]
+		bool HidesOthers { get; set; }
+
+		[Export ("forPrinting")]
+		bool ForPrinting { [Bind ("isForPrinting")] get; set; }
+
+		[Export ("createsNewApplicationInstance")]
+		bool CreatesNewApplicationInstance { get; set; }
+
+		[Export ("allowsRunningApplicationSubstitution")]
+		bool AllowsRunningApplicationSubstitution { get; set; }
+
+		[Export ("arguments", ArgumentSemantic.Copy)]
+		string[] Arguments { get; set; }
+
+		[Export ("environment", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, NSString> Environment { get; set; }
+
+		[NullAllowed, Export ("appleEvent", ArgumentSemantic.Strong)]
+		NSAppleEventDescriptor AppleEvent { get; set; }
+
+		// // Commenting this till issue #7304 is fixed
+		// [Internal]
+		// [Export ("architecture")]
+		// int _LaunchArchitecture { get; set; }
+
+		// CFBundle.Architecture LaunchArchitecture {
+		// 	[Wrap ("(CFBundle.Architecture) this._LaunchArchitecture")]
+		// 	get;
+		// 	[Wrap ("this._LaunchArchitecture = (int) value")]
+		// 	set;
+		// }
```

Comment:
Once https://github.com/xamarin/xamarin-macios/pull/7309 lands you can uncomment this. Will ping you when done.